.github/workflows/build-webapi.yaml

```diff
@@ -9,27 +9,45 @@ on:
     branches: [ dev, main ]
     paths:
       - 'src/dotnet/**'
+      - '.github/workflows/**'
 
 jobs:
-  build:
+  container-build-step:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: ./src/dotnet/CarbonAware.WebApi
     steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+    
+    - name: Docker Target Build
+      run: docker build --target=build . -f CarbonAware.WebApi/Dockerfile -t ca-api-build
+      working-directory: ./src/dotnet
+
+    - name: Docker Unit Test
+      run: docker run -w //src/ ca-api-build dotnet test
+      working-directory: ./src/dotnet
 
-    - uses: actions/checkout@v2
-    - name: Setup .NET
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 6.0.x
+  container-publish-step:
+      runs-on: ubuntu-latest
+      steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
 
-    - name: Restore dependencies
-      run: dotnet restore
+      - name: Docker Target Final
+        run: docker build . -f CarbonAware.WebApi/Dockerfile -t ca-api
+        working-directory: ./src/dotnet
+        
+      - name: Docker Run Container
+        run: |
+          docker run -d --name publish-container -p 8080:80 ca-api
+          docker container ls
+          
+      - name: Docker Wget in container
+        run: |
+          docker exec publish-container apt update
+          docker exec publish-container apt install -y wget
+          docker exec publish-container wget -t 5 --waitretry=5 0.0.0.0:80/emissions/bylocation?location=eastus
 
-    - name: Build
-      run: dotnet build --no-restore
+      - name: Docker Ping From Host
+        run: |
+          wget -t 5 --waitretry=5 0.0.0.0:8080/emissions/bylocation?location=eastus
```

Comment:
It feels unnecessarily brittle to hardcode an endpoint and path here that we know will change

Reply:
I disagree. We know the path will change, but for the moment, this is the only real validation we can do. If the path changes and the pipeline fails, we'll know that we've made a breaking change that needs to be fixed.

@dnastrain ?